networking/kafka-additional-listeners/README.rst

```diff
@@ -0,0 +1,193 @@
+Deploy Confluent for Kubernetes
+===============================
+
+#. Set up the Helm Chart:
+
+   ::
+
+     helm repo add confluentinc https://packages.confluent.io/helm
+
+
+#. Install Confluent For Kubernetes using Helm:
+
+   ::
+
+     helm upgrade --install operator confluentinc/confluent-for-kubernetes
+  
+#. Check that the Confluent For Kubernetes pod comes up and is running:
+
+   ::
+     
+     kubectl get pods
+
+========================================
+Review Confluent Platform configurations
+========================================
+
+You install Confluent Platform components as custom resources (CRs). 
+
+You can configure all Confluent Platform components as custom resources. In this
+tutorial, you will configure all components in a single file and deploy all
+components with one ``kubectl apply`` command.
+
+The entire Confluent Platform is configured in one configuration file:
+``$TUTORIAL_HOME/confluent-platform.yaml``
+
+In this configuration file, there is a custom Resource configuration spec for
+each Confluent Platform component - replicas, image to use, resource
+allocations.
+
+  
+=========================
+Deploy Confluent Platform
+=========================
+
+#. Deploy Confluent Platform with the above configuration:
+
+   ::
+
+     kubectl apply -f $TUTORIAL_HOME/confluent-platform.yaml
+     
+#. Check that all Confluent Platform resources are deployed:
+
+   ::
+   
+     kubectl get confluent
+
+#. Get the status of any component. For example, to check Kafka:
+
+   ::
+   
+     kubectl describe kafka
+
+========
+Validate
+========
+
+Deploy producer application
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that we've got the infrastructure set up, let's deploy the producer client
+app.
+
+The producer app is packaged and deployed as a pod on Kubernetes. The required
+topic is defined as a KafkaTopic custom resource in
+``$TUTORIAL_HOME/secure-producer-app-data.yaml``.
+
+The ``$TUTORIAL_HOME/secure-producer-app-data.yaml`` defines the ``elastic-0``
+topic as follows:
+
+::
+
+  apiVersion: platform.confluent.io/v1beta1
+  kind: KafkaTopic
+  metadata:
+    name: elastic-0
+    namespace: confluent
+  spec:
+    replicas: 3 # change to 1 if using single node
+    partitionCount: 1
+    configs:
+      cleanup.policy: "delete"
+      
+Deploy the producer app:
+
+``kubectl apply -f $TUTORIAL_HOME/producer-app-data.yaml``
+
+Validate in Control Center
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Use Control Center to monitor the Confluent Platform, and see the created topic and data.
+
+#. Set up port forwarding to Control Center web UI from local machine:
+
+   ::
+
+     kubectl port-forward controlcenter-0 9021:9021
+
+#. Browse to Control Center:
+
+   ::
+   
+     http://localhost:9021
+
+#. Check that the ``elastic-0`` topic was created and that messages are being produced to the topic.
+
+Review the additional listeners
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Review Kafka ConfigMap to check the additional custom listeners
```

Comment:
Instead of looking at the kafka ConfigMap, which is not what we want users to have to do, they can view the listeners configs through the Kafka CustomResource status. Could you change this section to use Kafka CR status please?

```
$ kubectl get kafka kafka -oyaml
## This is an example output when RBAC is deployed
listeners:
      external:
        advertisedExternalEndpoints:
        - b0.mydomain.com:9092
        - b1.mydomain.com:9092
        - b2.mydomain.com:9092
        authenticationType: mtls
        client: |-
          bootstrap.servers=kafka.mydomain.com:9092
          security.protocol=SSL
          ssl.keystore.location=/mnt/sslcerts/tls-kafka/keystore.p12
          ssl.keystore.password=<<jksPassword>>
          ssl.truststore.location=/mnt/sslcerts/tls-kafka/truststore.p12
          ssl.truststore.password=<<jksPassword>>
        externalAccessType: loadBalancer
        externalEndpoint: kafka.mydomain.com:9092
        internalEndpoint: kafka.confluent.svc.cluster.local:9092
        tls: true
      internal:
        authenticationType: mtls
        client: |-
          bootstrap.servers=kafka.confluent.svc.cluster.local:9071
          security.protocol=SSL
          ssl.keystore.location=/mnt/sslcerts/keystore.jks
          ssl.keystore.password=<<jksPassword>>
          ssl.truststore.location=/mnt/sslcerts/truststore.jks
          ssl.truststore.password=<<jksPassword>>
        internalEndpoint: kafka.confluent.svc.cluster.local:9071
        tls: true
      replication:
        authenticationType: mtls
        internalEndpoint: kafka.confluent.svc.cluster.local:9072
        tls: true
      token:
        authenticationType: oauthbearer
        internalEndpoint: kafka.confluent.svc.cluster.local:9073
        tls: true
```